store/index.js

```diff
@@ -306,11 +306,16 @@ export const actions = {
       return;
     }
     if (numBlocks === undefined) {
-      numBlocks = 10 * 60 * 24 * 7; // Look 7 days back
+      numBlocks = -(10 * 60 * 24 * 7); // Look 7 days back
+    } else if (numBlocks === "all") {
+      numBlocks = undefined;
```

Comment:
I think there's a codesmell of comparing numBlocks to undefined and then setting it to undefined. It should be two variables.

Frankly we don't ever search anything other than 7 days or all, so might as well be a boolean input called `searchAllHistory`

Reply:
Yea I agree it's a codesmell, but also wrapping different layers of interfaces and their defaults that don't necessarily make sense.

I don't think it makes sense to hide access to numBlocks though, it can come in handy to have more nuanced handling here in the future. Eh I'm inclined to leave it as-is until you mess with it next time you can change it then if you feel strongly.